src/core/utils/intersectUtils.js

```diff
@@ -67,14 +67,12 @@ export function intersectRay( nodeIndex32, array, ray ) {
 
 	if ( ( tmin > tzmax ) || ( tzmin > tmax ) ) return false;
 
-	// if ( tzmin > tmin || tmin !== tmin ) tmin = tzmin; // Uncomment this line if add the distance check
+	if ( tzmin > tmin || tmin !== tmin ) tmin = tzmin;
 
 	if ( tzmax < tmax || tmax !== tmax ) tmax = tzmax;
 
 	//return point closest to the ray (positive side)
 
-	if ( tmax < 0 ) return false;
-
-	return true;
+	return tmin <= far && tmax >= near; // TODO far and near should be scaled here
```

Comment:
`near` and `far` should be multiplied by `scale`. In case the scaling is not uniform, we need to have 3 separate values of `near` and `far` according to the axes. 
So `near` and `far` could become a Vector3 instead of a number. 
Do you have any ideas in mind otherwise?

Reply:
This should be validated further but I believe the math here works out such that if you scale the ray into the local space of the box being intersected against then the "distance" calculated here (tmin, tmax) is scaled based on the length of the ray. Ie tmin and tmax represent the number of ray.direction vectors needed to reach the surface. So this distance should be enough as-is.

Here's a modified version of the three.js intersects box function showing the above:

<details>
<summary>Code Block</summary>

```js
function intersectBox( box, ray ) {

    let tmin, tmax, tymin, tymax, tzmin, tzmax;

    const invdirx = 1 / ray.direction.x,
        invdiry = 1 / ray.direction.y,
        invdirz = 1 / ray.direction.z;

    const origin = ray.origin;

    if ( invdirx >= 0 ) {

        tmin = ( box.min.x - origin.x ) * invdirx;
        tmax = ( box.max.x - origin.x ) * invdirx;

    } else {

        tmin = ( box.max.x - origin.x ) * invdirx;
        tmax = ( box.min.x - origin.x ) * invdirx;

    }

    if ( invdiry >= 0 ) {

        tymin = ( box.min.y - origin.y ) * invdiry;
        tymax = ( box.max.y - origin.y ) * invdiry;

    } else {

        tymin = ( box.max.y - origin.y ) * invdiry;
        tymax = ( box.min.y - origin.y ) * invdiry;

    }

    if ( ( tmin > tymax ) || ( tymin > tmax ) ) return null;

    if ( tymin > tmin || isNaN( tmin ) ) tmin = tymin;

    if ( tymax < tmax || isNaN( tmax ) ) tmax = tymax;

    if ( invdirz >= 0 ) {

        tzmin = ( box.min.z - origin.z ) * invdirz;
        tzmax = ( box.max.z - origin.z ) * invdirz;

    } else {

        tzmin = ( box.max.z - origin.z ) * invdirz;
        tzmax = ( box.min.z - origin.z ) * invdirz;

    }

    if ( ( tmin > tzmax ) || ( tzmin > tmax ) ) return null;

    if ( tzmin > tmin || tmin !== tmin ) tmin = tzmin;

    if ( tzmax < tmax || tmax !== tmax ) tmax = tzmax;

    //return point closest to the ray (positive side)

    if ( tmax < 0 ) return null;

    return tmin >= 0 ? tmin : tmax;

}



box = new THREE.Box3();
box.min.setScalar( - 1 );
box.max.setScalar( 1 );

ray = new THREE.Ray();
ray.origin.set( 0, 0, - 5 );

ray.direction.set( 0, 0, 1 );
console.log( 'distance:', intersectBox( box, ray ) ); // 4

ray.direction.set( 0, 0, 0.5 );
console.log( 'distance:', intersectBox( box, ray ) ); // 8
```
</details>

Reply:
Scaling the direction of the ray works, but I need to understand if it can break intersections with triangles.

Reply:
I see now that [Ray.direction](https://threejs.org/docs/?q=ray#api/en/math/Ray.direction) is supposed to be normalized and the `Ray.applyMatrix4` function actually normalizes the ray rather than scaling it based on the matrix so my suggestion won't work.

Instead we can scale the near / far values based on how much the ray length would change which should work universally:

```js
const tmp = new Vector3().copy( ray.direction ).applyMatrix4( tmpInverseMatrix );

const scaleFactor = tmp.length() / ray.direction.length();
raycaster.near *= scaleFactor;
raycaster.far *= scaleFactor;
```

I _think_ this should work. We're basically just scaling the near and far values based on how the ray would scale to the local space. This has turned out to be more complicated than I though. Probably why it wasn't done like this in the first place 😅 Thanks for digging in and checking everything.

Reply:
There are still some problems... I will investigate later